### PR TITLE
feat(ElementInjector): Support injecting @Attributes()

### LINKED ISSUE #1818: Support a way to get all the attributes for a directive
In the existing material design library for Angular 1, there are some components that use a concept of "responsive attributes". Say you have a component like `md-grid-list`, which has an attribute `cols`. You can specify the value conditionally on screen size, for example:
```
cols-sm="2"     // Small screen size
cols-lt-lg="4"  // Less than large screen size
```

We currently have a way to inject a _single_ attribute (`@Attribute('cols')`), but the number of combinations gets large quickly (this one attribute has 9). One could also set up `properties`, but again it's a lot of boilerplate and unnecessary bindings. 

One solution would be to allow injection of a `<string, string>` attribute map. 

As a workaround, when per-component renderers are supported, the renderer can read the attributes and pass the values back to the directive via an event.

### PR DESCRIPTION
**Pull request by erictsangx**
Closes #1818

```
//get all attributes
class MyClass {
  constructor(@Attributes() atts: Map<string, string>)
}
```

### CODE REVIEW

**Comment by IgorMinar** [comment]
@vsavkin can you please review? at high level this looks alright to me.

**Comment by vicb** [comment]
Do we really want `@Attribute()` (vs `@Attributes`) here ?

File: modules/angular2/test/core/linker/element_injector_spec.ts [hunk 1]
```diff
@@ -140,6 +140,13 @@ class NeedsAttribute {
   }
 }
 
+class NeedsAllAttributes {
+  attributes: Map<string, string>;
+  constructor(@Attribute() attributes: Map<string, string>) {
```
Comment (IgorMinar) [inline modules/angular2/test/core/linker/element_injector_spec.ts]: as @vicb said, this should be `@Attributes()` no?

Comment (tbosch) [inline modules/angular2/test/core/linker/element_injector_spec.ts]: Could we wait with this PR until my Perf changes are in?

**Comment by tbosch** [comment]
We are in the process of complete rewriting the compiler to be based on codegen. Could you create an issue describing the feature?
Closing this PR as it also can't be rebased.

**Pull request closed without merging**